### PR TITLE
[INFINITY-2611] Enable tests to work correctly in strict mode clusters

### PR DESCRIPTION
**Pull request by NimaVaziri**
The main issue with this not working initially was performing operations in the `kdc` (and `hdfsclient` app) as `root` user, hence it working in permissive. When the same set of steps were performed, it obviously failed because the images (and the files/executables being accessed) weren't given the appropriate permissions/ownership for user `nobody`.

Since this now works in both strict and permissive, it's safe to say these pieces of the auth tooling are relatively stable now, so we should probably move the image to the `mesosphere` docker hub  account.

### CODE REVIEW

**Comment by NimaVaziri** [comment]
Real test of functionality is against this TC job using a strict cluster: https://teamcity.mesosphere.io/viewQueued.html?itemId=859672

**Comment by elezar** [comment]
@NimaVaziri were any changes to the client docker image required?

**Comment by benclarkwood** [comment]
@NimaVaziri Just realized (sorry): this test or the suite should be marked such that it's skipped in 1.9.

**Comment by NimaVaziri** [comment]
@elezar if by client docker image you're referring to `hdfs-client:dev` then yeah, there was a script placed in `/` that needed to be modified. Just making sure the paths and executables are accessible by `nobody` should be sufficient for the Kafka client image as well, I'd suppose.

@benclarkwood good catch, forgot about that

File: frameworks/hdfs/tests/test_auth.py [hunk 1]
```diff
@@ -78,21 +78,17 @@ def kerberos(configure_universe):
...
+@pytest.mark.dcos_min_version('1.10')
+@sdk_utils.dcos_ee_only
 @pytest.mark.smoke
 def test_health_of_kerberized_hdfs():
     config.check_healthy(service_name=config.SERVICE_NAME)
 
 
-sdk_utils.is_strict_mode()
+@pytest.mark.dcos_min_version('1.10')
+@sdk_utils.dcos_ee_only
```
Comment (elezar) [inline frameworks/hdfs/tests/test_auth.py]: Thanks for adding these to my PR too.

**Pull request merged**